src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.stream.IntStream;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LENGTH = 10;
+    private Object[] values;
+    private int size;
+
+    public ArrayList() {
+        values = new Object[DEFAULT_LENGTH];
+    }
+
     @Override
     public void add(T value) {
-
+        if (validateSize(size)) {
+            increaseArray();
+        }
+        values[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Wrong index out of bounds, please check index!");
+        }
+        if (validateSize(size)) {
+            increaseArray();
+        }
+        System.arraycopy(values, index, values, index + 1, size - index);
+        values[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        IntStream.range(0, list.size()).mapToObj(list::get).forEach(this::add);
```

Comment:
```suggestion
        for (T element : list) {
            add(element);
        }
```
i meant something like this :)